src/sinks/new_relic_logs.rs

```diff
@@ -61,6 +78,9 @@ impl SinkConfig for NewRelicLogsConfig {
 impl NewRelicLogsConfig {
     fn create_config(&self) -> crate::Result<HttpSinkConfig> {
         let mut headers: IndexMap<String, String> = IndexMap::new();
+        if self.encoding.codec != Encoding::Json {
+            error!("Using an unsupported encoding for New Relic")
```

Comment:
```suggestion
            return Err("Using an unsupported encoding for New Relic".into());
```

I think its better to funnel this through the error chain?

Reply:
Agreed!